### PR TITLE
[Refactoring] [TypeCheck] Get rid of 'dummy' nonsense

### PR DESCRIPTION
**Pull request by effectfully**
It only took like what, five years to kill this abomination?

### CODE REVIEW

File: plutus-core/untyped-plutus-core/test/Evaluation/Golden/polyError.plc.golden [hunk 1]
```diff
@@ -1 +1 @@
-(Right (delay (error )))
\ No newline at end of file
+(Right (delay (error)))
```
Comment (effectfully) [inline plutus-core/untyped-plutus-core/test/Evaluation/Golden/polyError.plc.golden:1]: Fixed that as a bonus.

**Review by kwxm — APPROVED** [review]
This looks fine.  I had a few quibbles but that was just because I felt obliged to say _something_.  It's certainly an improvement on what was there before.

File: plutus-core/plutus-core/src/PlutusCore/TypeCheck/Internal.hs [hunk 2]
```diff
@@ -350,7 +335,9 @@ inferKindM (TyApp ann fun arg)     = do
         KindArrow _ dom cod -> do
             checkKindM ann arg dom
             pure cod
-        _ -> throwing _TypeError $ KindMismatch ann (void fun) (KindArrow () dummyKind dummyKind) funKind
+        _ -> do
+            let expectedKindArrow = ExpectedShape "fun dom cod" ["dom", "cod"]
```
[the remark this answers is not in the document]
Comment (effectfully) [reply plutus-core/plutus-core/src/PlutusCore/TypeCheck/Internal.hs:339]: > Could we be in a context that might already involve a `k1` or `k2` that would mess that up?

No, `dom` and `cod` are meant to be existential, so "for some `dom` and `cod`" shadows whatever `dom` and `cod` we can have in the scope. Something about `k1` and `k2` makes me uncomfortable, how about we settle on `fun k l`?

File: plutus-core/plutus-core/src/PlutusCore/Error.hs [hunk 3]
```diff
@@ -186,24 +211,48 @@ instance ( Pretty ann
...
+--  for some 'a'
+-- >>> existentialVars ["'a'", "'b'"]
+--  for some 'a' and 'b'
+-- >>> existentialVars ["'a'", "'b'", "'c'"]
+--  for some 'a', 'b' and 'c'
+existentialVars :: [Doc ann] -> Doc ann
+existentialVars [] = ""
+existentialVars (x0:xs0) = " for some " <> go x0 xs0 where
+    go x []     = x
+    go x [y]    = x <> " and " <> y
```
[the remark this answers is not in the document]
Comment (effectfully) [reply plutus-core/plutus-core/src/PlutusCore/Error.hs]: OK-OK, Oxford comma it is!

**Comment by effectfully** [comment]
Comments addressed, @kwxm please rereview.

File: plutus-core/plutus-core/src/PlutusCore/Error.hs (hunk 3 above)
[the remark this answers is not in the document]
Comment (kwxm) [reply plutus-core/plutus-core/src/PlutusCore/Error.hs]: > OK-OK, Oxford comma it is!

You're probably better off without that:
```
existentialVars ["a","b"]
" for some a, and b"
```

[the remark this answers is not in the document]
Comment (effectfully) [reply plutus-core/plutus-core/src/PlutusCore/Error.hs]: Natural languages are stupid.

[the remark this answers is not in the document]
Comment (effectfully) [reply plutus-core/plutus-core/src/PlutusCore/Error.hs]: (So are programming ones)

**Comment by effectfully** [comment]
> I think it was fine before, but thanks for the updates anyway.

Error messages look much better now, I believe, thanks for the comments!

**Pull request merged**